sql/moz-fx-data-shared-prod/telemetry_derived/experiment_enrollment_aggregates_hourly_v1/metadata.yaml

```diff
@@ -0,0 +1,17 @@
+friendly_name: Experiment Enrollment Aggregates Hourly
+description: >
+  Experiment enrollment aggregate data from the events live table
+  from the past hour, partitioned by hour.
+labels:
+  incremental: true
+owners:
+- ascholtz@mozilla.com
+scheduling:
+  dag_name: bqetl_experiments_hourly
+  task_name: experiment_enrollment_aggregates_hourly
+  parameters: ["submission_timestamp:TIMESTAMP:{{ts}}"]
+  referenced_tables: [
+    "moz-fx-data-shared-prod", "telemetry_derived", "events_live"
+  ]
+  date_partition_parameter: submission_timestamp
```

Comment:
I don't think this is going to work without changes to `bigquery_etl_query` in telemetry-ariflow. The relevant code is:

```
    if destination_table is not None and date_partition_parameter is not None:
        destination_table = destination_table + "${{ds_nodash}}"
        parameters += (date_partition_parameter + ":DATE:{{ds}}",)
```

So it assumes a DATE type, and it injects `ds_nodash`. A hourly partition destination table has to be specified with a suffix of `$YYYMMDDHH`.

Do we have any other potential use cases queued up for hourly queries that might want to use a pattern like this? If not, it may be best to express this as a custom `query.py` that handles both the partitioning logic and the execution_delta. Instead of introducing the execultion_delta parameter to tasks, the task could execute immediately, and adjust the timestamp back 30 minutes. The DAG would run on the half hour in that case.

Reply:
We'll also need to set up an hourly query for monitoring experiment search aggregates. There is also some interest in monitoring engagement data, which would probably use the same mechanism, though nothing has been planned out for this use case yet.

Reply:
I think I might have found a workaround by using the Airflow macro [`ds_format`](https://airflow.apache.org/docs/apache-airflow/stable/macros-ref.html#airflow.macros.ds_format) which can format the nodash timestamp into a timestamp that indicates the hourly table partition:

```
    experiment_enrollment_aggregates_hourly = bigquery_etl_query(
        ...
        destination_table='experiment_enrollment_aggregates_hourly_v1${{ macros.ds_format(ts_nodash, "%Y%m%d%H%M%S", "%Y%m%d%H") }}',
        date_partition_parameter=None,
        ...
    )
```

The [GKEPodOperator `argument` field is a templated field](https://airflow.apache.org/docs/apache-airflow/1.10.12/_api/airflow/contrib/operators/kubernetes_pod_operator/index.html#module-airflow.contrib.operators.kubernetes_pod_operator), so when `destination_table` gets passed as argument it should get evaluated correctly.

Reply:
Do you think it would be possible to also solve the TimeDeltaSensor problem using `ds_format`?

We could have the DAG set to run on `HH:30` and then pass in parameter `submission_timestamp:TIMESTAMP:{{ macros.ds_format(ts, "%Y-%m-%d-%H:00:00") }}` (ts truncated to the hour). Then there would be no need to introduce a delay.

Reply:
Indeed! I updated the metadata.yaml for the query and removed TimeDeltaSensor